src/styles/index.scss

```diff
@@ -1,3 +1,63 @@
+$days-of-week: (
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+  sun: 6,
+);
+
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.calendar {
+  width: 707px;
+  display: flex;
+  flex-wrap: wrap;
+
+
+  &__day {
+    padding: 10px;
+    display: flex;
+    box-sizing: border-box;
+    height: 100px;
+    width: 100px;
+    background-color: #eee;
+    border: 1px solid black;
+    gap: 1px;
+    font-size: 30px;
+    align-items: center;
+    justify-content: center;
+    transition: 0.5s;
+  };
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: #ffbfcb;
+    transform: translate(0, -20px);
+  };
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      content: '#{$i}';
+    }
+  };
+}
+
+@for $i from 28 through 31 {
+  .calendar--month-length-#{$i} {
+  .calendar__day:nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
```

Comment:
The loop should be placed inside the `.calendar` block to follow the BEM structure more clearly and to avoid potential specificity issues.